src/http/routes/render/rate-limiter.ts

```diff
@@ -0,0 +1,40 @@
+import { FastifyInstance } from 'fastify'
+import fp from 'fastify-plugin'
+import fastifyRateLimit from '@fastify/rate-limit'
+import Redis from 'ioredis'
+import { getConfig } from '../../../config'
+
+const { rateLimiterDriver, rateLimiterRedisUrl, rateLimiterRenderPathMaxReqSec } = getConfig()
+
+export const rateLimiter = fp((fastify: FastifyInstance, ops: any, done: () => void) => {
+  fastify.register(fastifyRateLimit, {
+    global: true,
+    max: rateLimiterRenderPathMaxReqSec * 4,
+    timeWindow: 4 * 1000, //4s
+    continueExceeding: true,
+    nameSpace: 'image-transformation-ratelimit-',
+    redis: rateLimiterDriver === 'redis' ? new Redis(rateLimiterRedisUrl || '') : undefined,
+    keyGenerator: function (request) {
+      const ip = request.headers['x-real-ip'] || request.headers['x-client-ip'] || request.ip
```

Comment:
should the rate limiter be per (tenant, ip) instead of just ip if its running in multi tenant mode? that way we can have different rate limits per project (tier)

Reply:
Make sense!